### PR TITLE
Add API access logs to export and support deleting #6501

### LINKED ISSUE #6501: API key logs should be visible in export and deleted.
## Enhancement Request
When deleting a user, EDD should clear any API Log requests associated with that user's API Keys
screen shot 2018-04-10 at 12 40 53 pm

### Justification or use case
[GDPR](#6216 )

### PR DESCRIPTION
**Pull request by cklosowski**
Fixes #6501

Proposed Changes:
1. Adds API Access logs to the data exporter
2. Deletes API Access logs on erasure request.
3. Fixed an issue where no customer is found yet an empty table was being displayed for the customer record on export.

### CODE REVIEW

File: includes/privacy-functions.php [hunk 1]
```diff
@@ -478,52 +483,53 @@ function edd_register_privacy_exporters( $exporters ) {
...
+				'name'  => __( 'All Email Addresses', 'easy-digital-downloads' ),
+				'value' => implode( ', ', $customer->emails )
+			),
+		)
+	);
+
+	$agree_to_terms_time = $customer->get_meta( 'agree_to_terms_time' );
+	if ( ! empty( $agree_to_terms_time ) ) {
+		$export_data['data'][] = array(
+			'name' => __( 'Agreed to Terms' ),
```
Comment (SDavisMedia) [inline includes/privacy-functions.php]: Looks like this is already in `master`, but the text domain is missing.

File: includes/privacy-functions.php [hunk 2]
```diff
@@ -478,52 +483,53 @@ function edd_register_privacy_exporters( $exporters ) {
...
+			'value' => date_i18n( get_option( 'date_format' ) . ' H:i:s', strtotime( $customer->get_meta( 'agree_to_terms_time' ) ) )
+		);
+	}
+
+	$agree_to_privacy_time = $customer->get_meta( 'agree_to_privacy_time' );
+	if ( ! empty( $agree_to_privacy_time ) ) {
+		$export_data['data'][] = array(
+			'name' => __( 'Agreed to Privacy Policy' ),
```
Comment (SDavisMedia) [inline includes/privacy-functions.php]: Looks like this is already in `master`, but the text domain is missing.

File: includes/privacy-functions.php [hunk 3]
```diff
@@ -784,6 +790,78 @@ function edd_privacy_file_download_log_exporter( $email_address = '', $page = 1
...
+ * @param string $email_address The email address to look up api access logs for.
+ * @param int    $page          The page of logs to request.
+ *
+ * @return array
+ */
+function edd_privacy_api_access_log_exporter( $email_address = '', $page = 1 ) {
+	global $edd_logs;
+
+	$user = get_user_by_email( $email_address );
+	$log_query = array(
```
Comment (pippinsplugins) [inline includes/privacy-functions.php]: this should bail before the query if no user is found.

File: includes/privacy-functions.php (hunk 1 above)
Comment (cklosowski) [reply includes/privacy-functions.php]: @SDavisMedia yeah, this was part of that merging issue branches before merging master thing. I'll get those fixed up.

**Comment by cklosowski** [comment]
@pippinsplugins fixed.

**Pull request merged**